src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -603,6 +603,22 @@ def process_declarations(self, parent, nodes, arg_list):
                         elif "intent(inout)" in normalized_string:
                             interface = ArgumentInterface(
                                 ArgumentInterface.Access.READWRITE)
+                        elif "save" in normalized_string:
+                            # Variables declared with SAVE attribute inside a 
+                            # module, submodule or main program are implicitly
+                            # SAVE'd (see Fortran specification 8.5.16.4) so it
+                            # is valid to ignore the attribute in these
+                            # situations.
+                            if not isinstance(decl.parent.parent,
+                                              (Fortran2003.Module,
```

Comment:
We are searching for the immediate context e.g. Module->Submodule->Subroutine->Declaration what is important here is that the declaration is in the context of the subroutine, so I don't think we can go through `x.root` easily.

Reply:
I guess it could still fail if `x.parent == None` if process_declarations() is given directly the declaration without any context. I added a check for this scenario in the conditional statment.

Reply:
Thanks.